### DISCUSSION TITLE
Integration of camera c code with device service c

### DISCUSSION THREAD

**Discussion by Rakshithakrishna2** [edgexfoundry/community#27] [Q&A]
Hi all,

I am using EdgeX- 2.1 version of all services including **Device-sdk-c** . 
I have got to build an application of video streaming through web camera using device-sdk-c.

It would be really helpful, on clarrifying, if I could integrate a C code with device-sdk-c and edgex services.

Kindly guide me. Thank you.

**Comment by satyam1990** [comment]
Yes you can integrate C code with EdgeX C Device SDK since it is SDK written in **C Language**.

It would be helpful if you get familiar with basics of C SDK using C SDK documentation below:
https://docs.edgexfoundry.org/2.2/getting-started/Ch-GettingStartedSDK-C/

Once you have a fair understanding of the C SDK then you will hopefully get pointers on how to integrate it with your application.

**Reply by Rakshithakrishna2** [reply]
Yes, I have tried integrating with other sensors and even for capturing images from web camera.

Wanted to know if **VIDEO STREAMING** is possible with **device-sdk-c** ...?

Is there any sources available in C,  **not in GO**  ?

**Reply by satyam1990** [reply]
There are no API's in EdgeX C SDK which can help you do streaming, although there is an already existing EdgeX microservice named device-usb-camera which can do streaming from webcam. But this device service device-usb-camera might not have all features you may need.
